### ISSUE TITLE
SOLVED: Bug in TestDataset?

### ISSUE THREAD

**Issue by moisestg** [chenyuntc/simple-faster-rcnn-pytorch#107]
I decided to incorporate the traces for the validation loss. For that I created an `eval_loss()` function that basically does a forward pass without calculating any gradients and returning the loss. To load the validation data I used the `TestDataset` class as done in the already existing code. When I plotted the losses, the traces for the validation set didn't make any sense with very high values always increasing over time. After a long time, I discovered that the `TestDataset` class only resizes the images (according to the max and min size specified as parameters) but forgets to also resize the corresponding bounding boxes (this is done correctly for the `Dataset` class).

https://github.com/chenyuntc/simple-faster-rcnn-pytorch/blob/6a538bdd113bab649f388aa48193c1b1ccc48ee6/data/dataset.py#L125

Am I missing something here or this is just wrong? That would mean that all the metrics computed in `eval()` are incorrect. For the moment changing the class for loading the data is giving me validation traces that do make sense.

**Comment by moisestg** [comment]
I see now why the metrics are actually correct, the `faster_rcnn.predict()` function compensates for this fact and transforms the predicted bounding boxes back to the original image scale. 
https://github.com/chenyuntc/simple-faster-rcnn-pytorch/blob/6a538bdd113bab649f388aa48193c1b1ccc48ee6/model/faster_rcnn.py#L237
However the losses were incorrect for me as they were computed using bounding boxes with different scales.

**Comment by chenyuntc** [comment]
Sorry did not respond in time, and so glad you solve it.

**Comment by 300LiterPropofol** [comment]
> I see now why the metrics are actually correct, the `faster_rcnn.predict()` function compensates for this fact and transforms the predicted bounding boxes back to the original image scale.
> https://github.com/chenyuntc/simple-faster-rcnn-pytorch/blob/6a538bdd113bab649f388aa48193c1b1ccc48ee6/model/faster_rcnn.py#L237
> 
> 
> However the losses were incorrect for me as they were computed using bounding boxes with different scales.
Hi!
I bumped into the same issue and fixed it in the same way as you. Resize the bboxes before outputing validation loss. But my validation loss seems to keep at a relatively stable level and does not go down fast by the iterations, is this happen to you too? 
![image](https://user-images.githubusercontent.com/46748408/77735544-0b9cd480-700b-11ea-8d56-86b1ec82a82b.png)
